core/sys/linux/00_linux.odin

```diff
@@ -0,0 +1,144 @@
+
+package linux
+
+// Note(flysand): The filename starts with 0
+// because it needs to go first in alphabetical order
+// because it defines some constants used in other files
+
+/// Architecture to compile for
+ARCH :: ODIN_ARCH_STRING
+
+/// Bits per long
+when size_of(c.long) == 4 {
+	LONG_BITS :: 32
+} else {
+	LONG_BITS :: 64
+}
+
+// Aliases for commonly-used types
+uptr     :: uintptr
+when size_of(uptr) == 4 {
+	iptr :: i32
+} else when size_of(uptr) == 8 {
+	iptr :: i64
+}
+
+// Packages c and intrinsics are going to be used throughout this
+// package, so I'm re-exporting them in the current scope, so that
+// other files can see the identifiers.
+import "core:c"
+import "core:intrinsics"
+c_int    :: c.int
+c_uint   :: c.uint
+c_short  :: c.short
+c_ushort :: c.ushort
+c_long   :: c.long
+c_ulong  :: c.ulong
+c_llong  :: c.longlong
+c_ullong :: c.ulonglong
+
+/// This type is defined in this package rather than imported from core:c/libc
+/// because this package shouldn't be dependent on libc.
+c_timespec :: struct {
+	time_sec:  c_ulong,
+	time_nsec: c_ulong,
+}
+
+Time_Spec :: c_timespec
+
+Time_Val :: struct {
+	seconds: i64,
+	microseconds: int,
```

Comment:
Is `int` correct here?

Reply:
Technically yes, it's `i64` that's wrong. Though I think `c.long` is better for clarity